src/iden3comm/handlers/fetch.ts

```diff
@@ -219,11 +231,19 @@ export class FetchHandler
       PROTOCOL_MESSAGE_TYPE.CREDENTIAL_OFFER_MESSAGE_TYPE
     );
 
-    return this.handleOfferMessage(offerMessage, {
+    const message = await this.handleOfferMessage(offerMessage, {
```

Comment:
maybe private method `handleOfferMessage` should return `BasicMessage` and here in public method we cust to `CredentialIssuanceMessage`